enhancements/machine-config/manage-boot-images.md

```diff
@@ -0,0 +1,494 @@
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "managed-bootimages-platform-check"
+spec:
+  failurePolicy: Fail
+  paramKind:
+    apiVersion: config.openshift.io/v1
+    kind: Infrastructure
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["operator"]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["MachineConfiguration"]
+  validations:
+    - expression: "has(object.spec.MachineBootImageConfig) && param.status.platformStatus.Type != `GCP`"
+      message: "This feature is only supported on these platforms: GCP"
+```
+This would need an accompanying binding:
+```
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingAdmissionPolicyBinding
+metadata:
+  name: "managed-bootimages-platform-check-binding"
+spec:
+  policyName: "managed-bootimages-platform-check"
+  validationActions: [Deny]
+  paramRef:
+    name: "cluster"
+    namespace: "default"
+```
+#### Tracking boot image history
+
+This proposal will also introduce a new CR, `BootImageHistory` for tracking boot image history. As a starting point, here is a stub type definition for this:
+
+```
+type BootImageHistory struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   BootImageHistorySpec   `json:"spec,omitempty"`
+	Status BootImageHistoryStatus `json:"status,omitempty"`
+}
+
+// BootImageHistorySpec defines the desired state of BootImageHistory
+type BootImageHistorySpec struct {
+}
+
+// BootImageHistoryStatus defines the observed state of BootImageHistory
+type BootImageHistoryStatus struct {
+	// machineResourceReference contains identifying information of the machine management resource being tracked.
+	// +kubebuilder:validation:Required
+	// +required
+	MachineResourceReference MachineResourceReference `json:"machineResourceReference"`
+	// details is a list of boot image history entries of the machine resource.
+	// +optional
+	Details []BootImageHistoryDetail `json:"details,omitempty"`
+}
+
+type MachineResourceReference struct {
+	// name is the machine management resource's name
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+	// kind is the machine management resource's kind
+	// +kubebuilder:validation:Required
+	// +required
+	Kind string `json:"kind"`
+	// apiGroup is name of the APIGroup that the machine management resource belongs to. This is for disambiguating
+	// between Cluster API and Machine API backed resources.
+	// +kubebuilder:validation:Required
+	// +required
+	APIGroup string `json:"apiGroup"`
+}
+
+// BootImageHistoryDetail is the struct for each element in the Details array
+type BootImageHistoryDetail struct {
+	// updateTime records the timestamp at which the update took place.
+	// +required
+	UpdateTime metav1.Time `json:"updatedTime"`
+	// bootImageRef records the new boot image reference to which the update took place.
+	// +required
+	BootImageRef string `json:"bootImageRef"`
```

Comment:
If you're planning to stick with this (and not the configmap ref approach mentioned below), then I'd suggest adding a `Version` field as well to this object containing the actual RHCOS version string. In the case of GCP, it's part of the path, but that's not always the case depending on the cloud resource (e.g. AMIs).

Reply:
Understood, I'll be sure to include that on my next pass (: 

We'd still like to do the API-based approach for storing boot image history but it will be likely later in the pipeline and not part of the initial feature release, so totally open to iterating over this